service/worker/processor.go

```diff
@@ -267,54 +196,55 @@ func (p *replicationTaskProcessor) messageProcessLoop(workerWG *sync.WaitGroup,
 func (p *replicationTaskProcessor) processWithRetry(msg kafka.Message, workerID int) {
 	var err error
 
-	isInRetry := false
-	remainingRetryCount := retryCountInfinity
-	defer func() {
-		if isInRetry {
-			p.updateWorkerRetryStatus(false)
-		}
-	}()
+	forceBuffer := false
+	remainingRetryCount := p.config.ReplicationTaskMaxRetry
 
 ProcessRetryLoop:
-	for {
+	for attempt := 0; ; attempt++ {
 		select {
 		case <-p.shutdownCh:
 			return
 		default:
 			op := func() error {
```

Comment:
op should be move outside of the for loop

Reply:
Good point.  Moved outside.